### ISSUE TITLE
Auto project discovery not working on Mac OS with synced settings

### ISSUE THREAD

**Issue by diminutivesloop** [alefragnani/vscode-project-manager#164]
I’m currently running VSCode on both a Mac and Windows machine with my settings synced via the code-settings-sync extension. I’ve successfully configured auto project discovery on my Windows machine but on the Mac I keep getting the message “No projects saved yet!” no matter what I try. My user settings for this extension are as follows:
```json
"projectManager.sortList": "Recent",
"projectManager.git.baseFolders": [
    "/Users/nathanielpalmer/Development/",
    "G:\\projects"
],
"projectManager.vscode.baseFolders": [
    "/Users/nathanielpalmer/Development/",
    "G:\\projects"
],
"projectManager.git.maxDepthRecursion": 10,
"projectManager.vscode.maxDepthRecursion": 10,
```

**Comment by fredrikaverpil** [comment]
Not sure if related, but I'm on macOS and I can't get new projects to show even if I execute "Refresh Projects".

I can see all of my previous projects but no new projects appear.

**Comment by alefragnani** [comment]
Hi @diminutivesloop @fredrikaverpil ,

The MacOS/Windows sync should be working on the newer releases, because it's what [this changeset](https://github.com/alefragnani/vscode-project-manager/commit/9b8321d39829b02f140ea3251fb28d78460576e8) was created for, but maybe something has failed. 

I finally got a MacOS to test, so I will take a look if something was missed.

BTW, you could use `$home` _special variable_ while defining your `baseFolder`, as described in https://github.com/alefragnani/vscode-project-manager#automatic-detection-of-projects. It works for Windows/MacOS 😄 .

Hope this helps

**Comment by alefragnani** [comment]
I just tested on my MacOS and it appears to be working fine. I combined Windows and MacOS paths on the same setting, with and without projects in `projects.json` file.

My test settings was:

```
    "projectManager.git.baseFolders": [
       "$home\\Documents\\GitHub",
       "/Users/alefragnani/Documents/Bitbucket/",
       "G:\\Temp\\Projects"
    ],
```

I would ask you to check two things:

1. Make sure the paths are _case sensitive_, since MacOS is case sensitive
2. Look at `~/Library/Application Support/Code/User` folder and search for `projects.json` and `projects_cache_git.json` files. The `projects.json` file contains the projects that you _saved_ with `Project Manager: Save Project` command. The `projects_cache_git.json` file contains the projects that were _detected_ from your `projectManager.git.baseFolders` setting.

When you call `Project Manager: Refresh Projects` you will see the statusbar _loading_ the projects from each folder. The same _loading_ happens when you change the `projectManager` settings.

Please, let me know any news. If you still have some issue, I think I could add a _debug/verbose_ mode to try to detect what is wrong.

Hope this helps

**Comment by fredrikaverpil** [comment]
I'm on macOS and this is in my vscode `settings.json`:

```
    "projectManager.sortList": "Name",
    "projectManager.git.baseFolders": [
        "$home/code/repos"
    ],
```

This is where I store my projects on Windows/macOS:
- `C:\Users\fredrik\code\repos`
- `/Users/fredrik/code/repos`

And this is in my `projects.json`:

```
[
	{
		"name": "Windows only-project #1",
		"rootPath": "//10.0.1.111/pipeline/dev1",
		"paths": [],
		"group": ""
	},
	{
		"name": "Windows-only project #2",
		"rootPath": "//10.0.1.111/pipeline/dev2",
		"paths": [],
		"group": ""
	}
]

```

My `projects_cache_git.json` has a load of entries, most of repositories.

When I hit "Refresh projects", I do not see newly-cloned git folders in my `$home/code/repos` location, regardless of whether I'm attempting this on Windows or macOS.

**Comment by fredrikaverpil** [comment]
I just removed the `projects_cache_git.json` file, hit "Refresh projects" and no such file was re-created. And now I have no projects to select from in vscode.

**Comment by fredrikaverpil** [comment]
Ah, I also had to do a cmd+alt+enter to start indexing.

After the indexing, a new `projects_cache_git.json` was created which now also includes my newly cloned repositories... ok, weird.

**Comment by fredrikaverpil** [comment]
This happened again on macOS. Newly added project isn't found after refreshing.

**Comment by gor0n** [comment]
Same issue here on macOS. It was working fine before.

**Comment by andretf** [comment]
#172

**Comment by alefragnani** [comment]
🏃

**Comment by alefragnani** [comment]
Released in 0.24.1
